core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -6769,4 +6771,43 @@ private void checkJoinAssociateRuleWithTopAlwaysTrueCondition(boolean allowAlway
       relFn(relFn).with(hepPlanner).checkUnchanged();
     }
   }
+
+  /**
+   *  Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-4652">[CALCITE-4652]
+   *  AggregateExpandDistinctAggregatesRule must cast top aggregates to original type</a>.
+   *
+   *  Checks AggregateExpandDistinctAggregatesRule when return type of the SUM aggregate
+   *  is changed (expanded) by define custom type factory.
+   */
+  @Test void testDistinctCountWithExpandSumType() {
+    /* Expand SUM return type. */
+    RelDataTypeFactory typeFactory = new SqlTypeFactoryImpl(new RelDataTypeSystemImpl() {
```

Comment:
Use '//' not '/*' for comments.

"Expand SUM return type" doesn't adequately explain what you are doing here. You're creating a new type system. Say why.

Rather than creating a typeFactory, I'd create a type factory supplier. I'd assign the type system to a variable, to reduce nesting.